src/diffusers/modeling_flax_utils.py

```diff
@@ -394,6 +399,82 @@ def from_pretrained(
         # flatten dicts
         state = flatten_dict(state)
 
+        prng_key = prng_key if prng_key is not None else jax.random.PRNGKey(0)
+        params_shape_tree = jax.eval_shape(model.init_weights, prng_key)
+        required_params = set(flatten_dict(unfreeze(params_shape_tree)).keys())
+
+        random_state = flatten_dict(unfreeze(params_shape_tree))
```

Comment:
```suggestion
        shape_state = flatten_dict(unfreeze(params_shape_tree))
```

Reply:
handled in https://github.com/huggingface/diffusers/pull/513/commits/305a5448d1de9c2b4c45ee47b150d37f6919466f